src/pip/_internal/resolve.py

```diff
@@ -160,7 +160,7 @@ def _check_skip_installed(self, req_to_install):
 
         if not self._is_upgrade_allowed(req_to_install):
             if self.upgrade_strategy == "only-if-needed":
-                return 'not upgraded as not directly required'
+                return 'already satisfied, not checking version'
```

Comment:
Change this to `"already satisfied, not upgraded"`.

Reply:
Just to be sure we are on the same page here, wasn't the point of #5302 that the message gave the impression that an upgrade was possible rather than merely not even considered? I'm not sure the message you propose here is substantially better than the old one. However, your news fragment suggestion gave me an idea, what about `'already satisfied, skip upgrade'`? It's the same length as the one you propose, and it seems to me it conveys exactly the right message: updating the package is not considered at all.

Reply:
Anyway, I don't have strong feelings about it, so if you prefer your version, I can go for it. It still has the advantage of being shorter than the current message.

Reply:
`already satisfied, skipping upgrade` works for me. :)